src/io/msh_basereader.cc

```
@@ -86,57 +86,49 @@ std::vector<int> const & BaseMeshReader::get_element_vector(bool boundary_domain
-	    if (actual_header_.n_entities != n_entities) {
-	    	WarningOut().fmt("In file '{}', '{}' section for field '{}', time: {}.\nWrong number of entities: {}, using {} instead.\n",
-	                tok_.f_name(), data_section_name_, field_name, actual_header_.time, actual_header_.n_entities, n_entities);
-	        // actual_header.n_entities=n_entities;
-	    }
-
-	    if (n_components == 1) {
-	    	// read for MultiField to 'n_comp' vectors
-	    	// or for Field if ElementData contains only one value
-	    	size_of_cache = actual_header_.n_components;
+    if ( !it->second->is_actual(header.time, field_name) ) {
+	    // check that the header is valid - expected_n_entities
+	    if (header.n_entities != expected_n_entities) {
+	    	WarningOut().fmt("In file '{}', '{}' section for field '{}', time: {}.\nDifferent number of entities: {}, computation needs {}.\n",
+	                tok_.f_name(), data_section_name_, field_name, header.time, header.n_entities, expected_n_entities);
+	        // header.n_entities = expected_n_entities;
 	    }
-	    else {
-	    	// read for Field if more values is stored to one vector
-	    	size_of_cache = 1;
-	    	if (actual_header_.n_components != n_components) {
-	    		WarningOut().fmt("In file '{}', '{}' section for field '{}', time: {}.\nWrong number of components: {}, using {} instead.\n",
-		                tok_.f_name(), data_section_name_, field_name, actual_header_.time, actual_header_.n_components, n_components);
-	    		actual_header_.n_components=n_components;
-	    	}
-	    }
-
-    	(*element_data_values_)[field_name]
-					= std::make_shared< ElementDataCache<T> >(field_name, actual_header_.time, size_of_cache, n_components*n_entities);
-    	this->read_element_data(*(it->second), actual_header_, n_components, boundary_domain );
+        // check that the header is valid, try to correct n_components
+        if (header.n_components != expected_n_components) {
```

This should rather throw an exception.

---

done

---

and tested in unit test